drivers/xen/xenbus/xs_comms.c

```diff
@@ -612,13 +624,30 @@ int xs_comms_init(void)
 		      HYPERVISOR_start_info->store_mfn;
 #else
 #error "Unsupported architecture"
+#endif
+#else /* PVH && X86_64 */
+	UK_ASSERT(hvm_get_parameter(HVM_PARAM_STORE_EVTCHN, &evtchn) == 0);
+	UK_ASSERT(hvm_get_parameter(HVM_PARAM_STORE_PFN, &store_pfn) == 0);
```

Comment:
note that store_pfn/evtchn can be 0 in case there is no xenbus available which is a valid configuration even though it is going to be limited regarding available PV devices

we don't want to attempt mapping 0 in such cases